exercises/yacht/examples/success-standard/src/Yacht.hs

```diff
@@ -0,0 +1,59 @@
+module Yacht (yacht, Category(..)) where
+
+import Control.Monad (join)
+import Data.List (sortOn, sort, group)
+
+data Category = Ones
+              | Twos
+              | Threes
+              | Fours
+              | Fives
+              | Sixes
+              | FullHouse
+              | FourOfAKind
+              | LittleStraight
+              | BigStraight
+              | Choice
+              | Yacht
+
+yacht :: Category -> [Int] -> Int
+yacht Ones   = valueScore 1
+yacht Twos   = valueScore 2
+yacht Threes = valueScore 3
+yacht Fours  = valueScore 4
+yacht Fives  = valueScore 5
+yacht Sixes  = valueScore 6
+
+yacht FullHouse = sum
+                . join
+                . (\xs -> if length xs == 2 then xs else [])
```

Comment:
This is the that concerns me the most, but I couldn't figure out another way to do it.